src/app/[bookId]/components/book-review-form.tsx

```diff
@@ -0,0 +1,101 @@
+'use client'
+
+import React from 'react'
+import { useForm, Controller } from 'react-hook-form'
+
+import BooksAPI from '@/api/books-api'
+import { Button } from '@/components/ui/button'
+import { Textarea } from '@/components/ui/textarea'
+import useUserStore from '@/hooks/store/use-user-store'
+
+interface BookReviewFormProps {
+  bookId: string
+}
+
+const BookReviewForm = ({
+  bookId,
+  updateBook,
+}: BookReviewFormProps & { updateBook: () => void }) => {
+  const { user } = useUserStore()
+
+  const {
+    handleSubmit,
+    control,
+    setValue,
+    watch,
+    formState: { isSubmitting },
+  } = useForm({
+    defaultValues: {
+      content: '',
+      rating: 0,
+    },
+  })
+
+  const allFields = watch()
+  const rating = watch('rating')
+
+  const allFieldsFilled = Object.values(allFields).every((value) => value)
+
+  const onSubmit = async (data: { content: string; rating: number }) => {
+    if (!user) return
+    const review = {
+      userId: user._id,
```

Comment:
```suggestion
    const review = {
      userId: user!._id,
```
there is no need to check, if there is no user, there will be no button with submit

Reply:
Fixed